### ISSUE TITLE
For the Share-in with other backprop layers

### ISSUE THREAD

**Issue by sunformoon** [PrincetonVision/marvin#50]
Hi I checked that the Softmax layer cannot share the in's with other backprop layers. I am wondering what does the outcome look like if I mistakenly share in's for Softmax layer?

Could I write a SplitLayer (like Caffe) to avoid this problem?

Thank you!
